### DISCUSSION TITLE
How to get current 'index' of snapped scroll item in a mandatory scroll-snap carousel

### DISCUSSION THREAD

**Discussion by stijnwesterhof** [tailwindlabs/tailwindcss#6450] [Help]
Hi all,

I'm trying to build a (react) carousel with the new scroll-snap utility. I have it working but I want to improve the UX by showing some dots like you see in most carousels.

I was wondering if anyone knew a simple, but smart solution on getting some sort of index of what image is currently 'snapped'.

I hope my question is clear :)

**Comment by philwolstenholme** [comment]
I think this is probably more of a JavaScript rather than Tailwind question, but my first thought was to do something with the Intersection Observer API: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

**Reply by philwolstenholme** [reply]
[https://24ways.org/2019/beautiful-scrolling-experiences-without-libraries](https://24ways.org/2019/beautiful-scrolling-experiences-without-libraries/#:~:text=through%20the%20sections%3A-,The%20Intersection%20Observer%20API,listening%20for%20scroll%20events.,-We%20can%20create) might be useful, or perhaps https://www.rubensuet.com/intersectionObserver/

**Reply by stijnwesterhof** [reply]
These blogs are very helpful! Thank you sir :)

**Reply by adrianhorning08** [reply]
@stijnwesterhof did you ever figure this out? I've been trying to figure this out for 2 hours, still nothing 😅

**Reply by MuhammadTanveerS** [reply]
Did anyone figure it out? I wanna know as well

**Reply by MentalGear** [reply]
This shows you how it's done:
https://stackoverflow.com/questions/53355384/indicators-dots-with-css-scroll-snap
